### PR TITLE
feat: creating a stripe sdk for simplified mocking

### PR DESCRIPTION
**Pull request by joelhooks**
[⭐️ long 45m end-to-end loom video of creating this PR](https://www.loom.com/share/28d14ea87b7b410092abd70d74d8bcb6)

Screen Shot 2022-11-08 at 10 14 09

we are exploring how to mock out services to test some of our commerce workflows. In this PR, I looked at how we were mocking `prisma` in the `@skillrecordings/database` package utilizing a database SDK that wraps `prisma` in a lightweight typed mockable context using `jest-mock-extended`

This PR introduces `@skillrecordings/stripe-sdk` which provides a similar mockable context for testing stripe without using a heavier solution for now using simple data objects as mocks.

We should consider moving **all `stripe` calls** into this SDK instead of accessing the `stripe` library across the entire application and package surface area.

IMO, this is true for these service boundaries in general. They get wrapped.

wrap it up

### CODE REVIEW

File: packages/commerce-server/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
     "@casl/ability": "^6.0.0",
     "@skillrecordings/database": "workspace:*",
+    "@skillrecordings/stripe-sdk": "workspace:*",
```
Comment (joelhooks) [inline packages/commerce-server/package.json:20]: _a new package has entered chat_

File: packages/commerce-server/src/record-new-purchase.ts [hunk 2]
```diff
@@ -29,16 +29,9 @@ export class PurchaseError extends Error {
...
-      expand: [
-        'customer',
-        'line_items.data.price.product',
-        'payment_intent.charges',
-      ],
-    },
-  )
+  const {getCheckoutSession} = getStripeSdk()
+
+  const checkoutSession = await getCheckoutSession(checkoutSessionId)
```
Comment (joelhooks) [inline packages/commerce-server/src/record-new-purchase.ts:34]: we should refactor to this style for accessing service APIs through 2nd/3rd party libs imo

File: packages/stripe-sdk/src/stripe-sdk.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      return await ctx.stripe.checkout.sessions.retrieve(checkoutSessionId, {
+        expand: [
+          'customer',
+          'line_items.data.price.product',
+          'payment_intent.charges',
+        ],
+      })
+    },
+  }
+}
```
Comment (joelhooks) [inline packages/stripe-sdk/src/stripe-sdk.ts:19]: by default the context carries the normal `stripe` client instance, but we can also mock it

not I didn't write a test and exercise the mock context to see if it works as expected, but hand rolling a mock is also simple and can **iterate into end game automated solution** as needed (versus aiming for that now)

**Review by jbranchaud — APPROVED** [review]
🚀

**Pull request merged**